src/modules/interfaces.ts

```diff
@@ -337,6 +342,32 @@ export class Module {
         install: migrationClass.prototype.up,
         remove: migrationClass.prototype.down,
       };
+      const syncified = new Function(
+        'return ' +
+        migrationClass.prototype.up
+          .toString()
+          .replace(/\basync\b/g, '')
+          .replace(/\bawait\b/g, '')
+          .replace(/^/, 'function')
+          // The following are only for the test suite, but need to be included at all times
+          .replace(/\/* istanbul ignore next *\//g, '')
+          .replace(/cov_.*/g, '')
+      )();
+      const tables: string[] = [];
+      const functions: string[] = [];
+      syncified({ query: (text: string) => {
```

Comment:
This looks good but I think my question is why is this inside the condition `if (!def.migrations) {`? shouldn't be always there?

Reply:
So nothing went through the other branch, but also if anything did, it would imply that the migrations don't exist as a separate file and this trick would not work.

In any case in the next PR we decided to just not allow "unconventional" migration definitions so the other branch no longer exists.